# 6번문제

## A학교 남녀 학생들의 평균 혈압 차이가 있는지 여부에 대한 검정 문제 

* 단, 남학생과 여학생의 혈압 데이터는 정규분포를 따르며 등분산임을 가정한다


> 1) 남녀 학생들의 평균 혈압 차이가 있는지에 대해 가설을 설정하라 

> 2) 검정통계량을 구하고 판단해라 

> 3) 평균 혈압차의 신뢰구간을 구했을 때, 판단한 결과가 2) 의 결과를 지지하는지 설명해라 

## 1) 남녀 학생들의 평균 혈압 차이가 있는지에 대해 가설을 설정하라

### 귀무가설 : A학교에서 성별에 따른 평균 혈압 차이는 없다. 
### 대립가설 : A학교에서 성별에 따른 평균 혈압 차이는 있다. 

In [2]:
import pandas as pd 
data = pd.read_csv("https://raw.githubusercontent.com/ADPclass/ADP_book_ver01/main/data/26_problem6.csv")
data.head(3)

gender  pressure
0   male     106.8
1   male     100.8
2   male      84.5

In [4]:
print(''' 
이 문제는 남학생 25명의 혈압 평균과 여학생 25명의 혈압 평균에 유의미한 차이가 있는 지 검정하는 문제이다.
      샘플의 수가 30개 미만이므로 독립집단 t-test를 수행한다.
''')

from scipy.stats import ttest_ind

stat = ttest_ind(data[data.gender=='male'].pressure, data[data.gender=='female'].pressure, equal_var=True)
print(f't검정통계량은 {stat.statistic:.3f}이고 pvalue는 {stat.pvalue:.3f}이다.')
print('유의수준 0.05에서 귀무가설을 기각할 수 없으며, 따라서 남학생과 여학생의 혈압에는 유의미한 차이가 없다.')



 
이 문제는 남학생 25명의 혈압 평균과 여학생 25명의 혈압 평균에 유의미한 차이가 있는 지 검정하는 문제이다.
      샘플의 수가 30개 미만이므로 독립집단 t-test를 수행한다.

t검정통계량은 1.381이고 pvalue는 0.180이다.
유의수준 0.05에서 귀무가설을 기각할 수 없으며, 따라서 남학생과 여학생의 혈압에는 유의미한 차이가 없다.


In [2]:
# 성별에 따른 혈압이 등분산성을 만족한다는 조건 하에 독립 t검정 수행 (ttest_ind), default is 'two-sided' 
import scipy.stats as stats

## 데이터 나누기 
male = data.loc[data.gender == 'male', 'pressure']
female = data.loc[data.gender == 'female', 'pressure']

test_result = stats.ttest_ind(male, female, equal_var=True)
test_result

Ttest_indResult(statistic=1.3813481801194591, pvalue=0.18044550626193734)

### 검정 결과: 귀무가설 기각하지 못한다. 
### 즉, 성별에 따라 평균 혈압의 차이가 없다고 할 수 있다. 

# 3) 평균 혈압차의 신뢰구간을 구했을 때, 판단한 결과가 2) 의 결과를 지지하는지 설명해라

In [7]:
print('''평균 혈압차의 신뢰구간이란, 주어진 유의수준에서 남학생/여학생 평균 혈압 차이가 어느 범위 밖을 벗어나면 서로 다르다고 결정하게 되는데, 이 범위를 말한다.
      - 남학생/여학생을 대상으로 여러 차례 샘플링하여 그 차이를 구했을 때, 
      - 남학생/여학생 모집단에 유의미한 차이가 없다면, 그 샘플 간 차이는 0을 중심으로 하는 정규분포를 이룰 것이다.

      - 유의수준 5%라면, 양측 검정 시 하위 0.025, 상위 0.025를 제외한 확률 구간이 된다.
''')

from scipy.stats import t

t.ppf(0.025,)

평균 혈압차의 신뢰구간이란, 주어진 유의수준에서 남학생/여학생 평균 혈압 차이가 어느 범위 밖을 벗어나면 서로 다르다고 결정하게 되는데, 이 범위를 말한다.
      - 남학생/여학생을 대상으로 여러 차례 샘플링하여 그 차이를 구했을 때, 
      - 남학생/여학생 모집단에 유의미한 차이가 없다면, 그 샘플 간 차이는 0을 중심으로 하는 정규분포를 이룰 것이다.

      - 유의수준 5%라면, 양측 검정 시 하위 0.025, 상위 0.025를 제외한 확률 구간이 된다.



TypeError: _parse_args() missing 1 required positional argument: 'df'

In [4]:
male.mean() - female.mean()

5.296527777777783

In [5]:
#평균과 신뢰구간
from numpy import array,mean
from scipy.stats import sem,t
import numpy as np


In [6]:
test_result.statistic

1.3813481801194591

In [7]:
t._ppf(0.025,(len(male)+len(female)-2)) ## 작은 값쪽 기각역 
## t분포를 따르는 값 _ppf ( a/2 , n1 -1 +  n2 -1 )

-2.068657610419041

In [8]:
## 기각역 : [-2.068,2.068]


In [9]:
## 신뢰구간은?? 결국 기각역에 Sp/sqrt(1/n1 + 1/n2) 를 해주면 된다. 


## 신뢰구간이 필요한건, 변환을 시켜서 가져가는 것이 아닌, 원본 데이터의 구간이 궁금!!!! 

In [10]:
## 통합 표준편차를 구하는 공식이 복잡하므로 함수를 만들어 사용 
def sp(data1, data2) : 
    df = len(data) + len(data2) -2
    s1 = (len(data1)-1) * (data1.std()**2)
    s2 = (len(data2)-1) * (data2.std()**2)
    sp = np.sqrt((s1+s2)/ df)
    return sp 
    

In [11]:
sp(male,female)

7.801682371761947

In [12]:
ci_v = t._ppf(0.025,(len(male)+len(female)-2)) * sp(male,female) * (1/len(male) + 1/len(female))
ci_v

-2.801911391044692

In [13]:
## 이제 우리의 관심은 다시 male.mean() - female.mean()

In [14]:
(male.mean() - female.mean())/sp(male,female) * (1/len(male) + 1/len(female))

0.1178638181758219

In [15]:
male.mean() - female.mean() - t._ppf(0.025,(len(male)+len(female)-2)) * sp(male,female) * (1/len(male) + 1/len(female))

8.098439168822475

In [16]:
male.mean() - female.mean() + t._ppf(0.025,(len(male)+len(female)-2)) * sp(male,female) * (1/len(male) + 1/len(female))

2.494616386733091

In [17]:
alpha = 0.95 
dof = len(male) + len(female) -2
diff_mean = male.mean() - female.mean()
s =  sp(male,female) * (1/len(male) + 1/len(female))

CI = t.interval(alpha, dof, loc = diff_mean, scale = s) 
CI

(2.4946163867330915, 8.098439168822473)

## 신뢰구간 : [2.495 , 8.098]  
* 소숫점 4째 자리 반올림 

우리가 검정에 사용했던 남자평균 - 여자평균의 값이 5.29*** 이었으므로 신뢰구간을 구한 결과도 앞선 독립 2표본 t검정 결과를 지지한다. 
